packages/compass-indexes/src/components/create-index-form/create-index-form.tsx

```diff
@@ -79,13 +80,40 @@ function CreateIndexForm({
       });
   }, [schemaFields]);
 
+  const showIndexesGuidanceIndexFlow =
+    showIndexesGuidanceVariant && currentTab === 'IndexFlow';
+  const showIndexesGuidanceQueryFlow =
+    showIndexesGuidanceVariant && currentTab === 'QueryFlow';
+
+  const RenderCreateIndexFields = () => {
+    if (fields.length > 0 && !showIndexesGuidanceQueryFlow) {
+      return (
+        <CreateIndexFields
+          schemaFields={schemaFieldNames}
+          fields={fields}
+          serverVersion={serverVersion}
+          isRemovable={!(fields.length > 1)}
+          onSelectFieldNameClick={onSelectFieldNameClick}
+          onSelectFieldTypeClick={onSelectFieldTypeClick}
+          onAddFieldClick={onAddFieldClick}
+          onRemoveFieldClick={onRemoveFieldClick}
+        />
+      );
+    }
+    return null;
+  };
```

Comment:
Defining this as is will cause it to be re-rendered on every render. That can be a significant performance issue when props change often. I'd suggest moving this similar to how it was before where we're rendering the `CreateIndexFields` in two places. It'll be a bit of duplication, but hopefully not too much as it's just one component.